### PR TITLE
Add instance property to types of ServerStyleSheet

### PR DESCRIPTION
**Pull request by bakunyo**
I use styled-component with TypeScript and [react_on_rails](https://github.com/shakacode/react_on_rails) .

I'd like to use `instance` property of ServerStyleSheet. But this type is not defined in typings.
reference:  https://github.com/shakacode/react_on_rails/issues/883#issuecomment-315431517

When a instance is set in ServerStyleSheet, that is a instance of ServerStyleSheet.
It's my understanding. So I added.

Please let me know if I mistake something.
Until this pr will be merged, I use below line in my typescript code.

```
const sheet = new ServerStyleSheet() as any;
```

### CODE REVIEW

File: typings/styled-components.d.ts [hunk 1]
```diff
@@ -112,6 +112,7 @@ export class ServerStyleSheet {
 
   getStyleTags(): string;
   getStyleElement(): ReactElement<{}>[];
+  instance: ServerStyleSheet;
```
Comment (kitten) [inline typings/styled-components.d.ts]: This is actually `StyleSheet` :wink:

File: test-results.json [hunk 2]
```diff
@@ -1 +1 @@
...
...components/src/utils/test/escape.test.js","startTime":1517970999855,"status":"passed","summary":""},{"assertionResults":[{"failureMessages":[],"status":"passed","title":"should take precedence over ThemeProvider"},{"failureMessages":[],"status":"passed","title":"should fallback to default theme"}],"endTime":1517970999936,"message":"","name":"/Users/bakunyo/workspace/github.com/bakunyo/styled-components/src/utils/test/determineTheme.test.js","startTime":1517970999837,"status":"passed","summary":""},{"assertionResults":[{"failureMessages":[],"status":"passed","title":"should create alphabetic names for number input data"}],"endTime":1517971000068,"message":"","name":"/Users/bakunyo/workspace/github.com/bakunyo/styled-components/src/utils/test/generateAlphabeticName.test.js","startTime":1517970999977,"status":"passed","summary":""},{"assertionResults":[{"failureMessages":[],"status":"passed","title":"should have all valid HTML5 elements defined as properties"}],"endTime":1517971000229,"message":"","name":"/Users/bakunyo/workspace/github.com/bakunyo/styled-components/src/constructors/test/styled.test.js","startTime":1517971000056,"status":"passed","summary":""}],"wasInterrupted":false}
```
Comment (kitten) [inline test-results.json]: This won't need to be commited (it's only still in the repo accidentally due to PRs readding it)

But we do need an entry to the changelog

**Comment by bakunyo** [comment]
@kitten Thanks for your review.
I have added two commits reflected your comments.
Could you please review this pr again.

**Comment by kitten** [comment]
@bakunyo sure, mate! Looks like it does the trick :+1: thanks for catching this. must've been overlooked

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Pull request merged**